crates/oxc_semantic/src/symbol/table.rs

```diff
@@ -102,10 +117,14 @@ impl SymbolTable {
         for reference in references {
             let resolved_reference_id =
                 ResolvedReferenceId::new(self.resolved_references.len() + 1);
+            let reference_span = reference.span;
+
             let resolved_reference = reference.resolve_to(symbol_id);
             self.resolved_references.push(resolved_reference);
             // explicitly push to vector here in correspondence to the previous reserve call
             symbol.references.push(resolved_reference_id);
+
+            self.resolved_references_index.insert(reference_span, resolved_reference_id);
```

Comment:
Did a simple experiment on https://unpkg.com/babylonjs@3.2.0-alpha4/babylon.max.js

Commenting out this line:
<img width="503" alt="Screen Shot 2023-04-08 at 1 23 09 AM" src="https://user-images.githubusercontent.com/42916814/230706820-31a7ea7e-f473-4bcf-9fa2-f76d4b42d797.png">

With this line:

<img width="490" alt="Screen Shot 2023-04-08 at 1 22 26 AM" src="https://user-images.githubusercontent.com/42916814/230706800-73d1864d-d89a-4408-a33a-51fbcb9f26d4.png">